internal/kubernetes/nodefilters.go

```diff
@@ -41,6 +42,22 @@ func NewNodeLabelFilter(labels map[string]string) func(o interface{}) bool {
 // NewNodeConditionFilter returns a filter that returns true if the supplied
 // object is a node with any of the supplied node conditions.
 func NewNodeConditionFilter(ct []string) func(o interface{}) bool {
+	var specifiedConditions [][]string
```

Comment:
It seems like we're using a `[]string` here to represent each condition key/value tuple. I'd strongly prefer we used a struct with named fields - we could reuse the existing `NodeCondition` type and set only the `Type` and `Status` fields - https://godoc.org/k8s.io/api/core/v1#NodeCondition.